Bottom: 85bf2b382d52b3e5f085c8c5eef712ff5e814816
Top:    5e077f665ca1a4c1ae8a0c52436f88b4bd220f59
Author: Alexandre Oliva <aoliva@redhat.com>
Date:   2016-03-09 11:51:45 -0300

Refresh of template-support.patch

---

diff --git a/libcc1/libcp1plugin.cc b/libcc1/libcp1plugin.cc
index dc3767e..7cfd075 100644
--- a/libcc1/libcp1plugin.cc
+++ b/libcc1/libcp1plugin.cc
@@ -1267,10 +1267,9 @@ plugin_start_new_template_decl (cc1_plugin::connection *self ATTRIBUTE_UNUSED)
 }
 
 gcc_typedecl
-plugin_type_decl (cc1_plugin::connection *self,
+plugin_type_decl (cc1_plugin::connection *,
 		  gcc_type type_in)
 {
-  plugin_context *ctx = static_cast<plugin_context *> (self);
   tree type = convert_in (type_in);
 
   tree name = TYPE_NAME (type);
@@ -1647,11 +1646,10 @@ plugin_type_value_expr (cc1_plugin::connection *self,
 }
 
 gcc_typedecl
-plugin_expr_type (cc1_plugin::connection *self,
+plugin_expr_type (cc1_plugin::connection *,
 		  gcc_expr operand)
 {
-  plugin_context *ctx = static_cast<plugin_context *> (self);
-  tree op0 = convert_in (operand1);
+  tree op0 = convert_in (operand);
   tree type = TREE_TYPE (op0);
   if (type)
     type = TYPE_NAME (type);
